ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -776,6 +776,18 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 			return err
 		}
 
+		// Delete all workers if worker group is suspended and skip reconcile
```

Comment:
a follow up issue: should we call `deleteAllPods` only when there are no in-flight requests? If so, I will add a follow up item in https://github.com/ray-project/kuberay/issues/2566.

Reply:
update: the expectation has already been checked by the following logic

```
if !r.rayClusterScaleExpectation.IsSatisfied(ctx, instance.Namespace, instance.Name, worker.GroupName) {
```

Reply:
maybe another follow up is to update `expectations` for `deleteAllPods` instead. Currently, we rely on `deleteAllPods` to check the number of Pods is more than 0 to avoid calling `DeleteAllOf` every reconciliation.

Reply:
Yeah I wasn't sure what to do about expectations, but it doesn't seem like we do anything for expectations in the standard suspend case with `.spec.suspend`

Reply:
> but it doesn't seem like we do anything for expectations in the standard suspend case with .spec.suspend

We need to address it in the future. You can check #2566 for more details, but I think it is fine not to handle expectations in this PR.